### PR TITLE
Implement clock

### LINKED ISSUE #18: Implement clock
Required for #12 and the logger.

The clock will be a component of the controller.
It will expose two methods to get and set the current date and time.

If we want the application to run faster than 1x time, care should be taken for the calculation of variations such as consumption, temperature, etc.
This logic should be separated from the "physical" clock component of the controller, for eg. in a specific `World` class (or in the simulation/emulation manager) which tracks the time and sets the controller clock as time passes.
The variations in consumptions, temperature, etc. can then be calculated with the step time (the delta time, the difference of time between the updates).

Implementation-wise Java provides a class for this [java.time.LocalDateTime](https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html).

### PR DESCRIPTION
**Pull request by smagnaniuni**
This class can be used to model the time and can be updated from the simulation side with a variable step.
It can be used as the internal clock of the controller but also on the simulation side.

Closes #18

### CODE REVIEW

File: app/src/main/java/homer/common/time/Clock.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+     * Sets the clock to a new local date and time.
+     * 
+     * @param dateTime The new local date and time to set.
+     */
+    void setDateTime(LocalDateTime dateTime);
+
+    /**
+     * Updates the clock local date and time with a delta time amount.
+     * 
+     * @param deltaTime How much time should be added to the current clock time.
```
Comment (michelececcacci) [inline app/src/main/java/homer/common/time/Clock.java:24]: what is the time's measure unit? i think it should be specified in the interface

Comment (smagnaniuni) [reply app/src/main/java/homer/common/time/Clock.java:24]: Do you mean I have to specify it's a [TemporalAmount](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/temporal/TemporalAmount.html)?
You can give for eg. a [Duration](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/Duration.html) or [Period](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/Period.html), which can be created in several ways.

**Review by aleemont1 — APPROVED** [review]
LGTM

File: app/src/main/java/homer/common/time/Clock.java (hunk 1 above)
Comment (smagnaniuni) [reply app/src/main/java/homer/common/time/Clock.java:24]: @michelececcacci not sure what you meant

Comment (michelececcacci) [reply app/src/main/java/homer/common/time/Clock.java:24]: oh, ok, perfect. there was a misunderstanding on my side.

**Pull request merged**